packages/angular/ssr/src/app.ts

```diff
@@ -113,3 +97,24 @@ export class AngularServerApp {
     return render(this, request, serverContext, requestContext);
   }
 }
+
+let angularServerApp: AngularServerApp | undefined;
+
+/**
+ * Creates an instance of `AngularServerApp` if it does not already exist.
+ * If an instance exists, it will return the existing one.
+ *
+ * @param options - (Optional) The configuration options for the Angular server app.
+ * @returns The created or existing instance of `AngularServerApp`.
+ */
+export function createAngularServerApp(options?: AngularServerAppOptions): AngularServerApp {
+  return (angularServerApp ??= new AngularServerApp(options));
+}
+
+/**
+ * Resets the instance of `AngularServerApp` to undefined, effectively
+ * clearing the reference. Use this to recreate the instance.
+ */
+export function resetAngularServerApp(): void {
+  angularServerApp = undefined;
+}
```

Comment:
nit: is there a need for any cleanup once reset happens? Another quick nit: may be call this function `destroyAngularServerApp` (so it pairs with "create" a bit better)?

Reply:
I am fine with destroy, no there is no additional cleanup required once reset happens.